### PR TITLE
CS: use nowdoc instead of heredoc

### PR DESCRIPTION
**Pull request by gharlan**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | — |
| License | MIT |
| Doc PR | — |

if this is accepted, we could add the fixer to symfony level of php-cs-fixer: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1580

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Fixtures/Resources/views/translation.html.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 This template is used for translation message extraction tests
 <?php echo $view['translator']->trans('single-quoted key') ?>
 <?php echo $view['translator']->trans('double-quoted key') ?>
-<?php echo $view['translator']->trans(<<<EOF
+<?php echo $view['translator']->trans(<<<'EOF'
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Fixtures/Resources/views/translation.html.php]: this must be reverted. It is precisely testing the extraction from heredoc (extracting from nowdoc is tested 3 lines later)

**Comment by nicolas-grekas** [comment]
Personally, I'm `-0` on this. I see no value here in comparison to the merge conflicts and added pressure on authors...

**Comment by gharlan** [comment]
We already force single quotes instead of double quotes (when possible). So in my opinion it would be consistent to do it with nowdoc vs. heredoc in the same way.

File: src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php [hunk 2]
```diff
@@ -106,16 +106,16 @@ private function generateDeclaredRoutes()
...
-            throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', \$name));
+        if (!isset(self::$declaredRoutes[$name])) {
+            throw new RouteNotFoundException(sprintf('Unable to generate a URL for the named route "%s" as such route does not exist.', $name));
         }
 
-        list(\$variables, \$defaults, \$requirements, \$tokens, \$hostTokens) = self::\$declaredRoutes[\$name];
+        list($variables, $defaults, $requirements, $tokens, $hostTokens) = self::$declaredRoutes[$name];
 
-        return \$this->doGenerate(\$variables, \$defaults, \$requirements, \$tokens, \$parameters, \$name, \$referenceType, \$hostTokens);
+        return $this->doGenerate($variables, $defaults, $requirements, $tokens, $parameters, $name, $referenceType, $hostTokens);
```
Comment (keradus) [inline src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php]: so cleaner with nowdoc ;)

**Comment by nicolas-grekas** [comment]
:+1: (not a big one but still :) )

**Comment by stof** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @gharlan.

**Comment by nicolas-grekas** [comment]
I merged 2.3 up to master without looking for more heredoc to change, which means they may be more in 2.7 and up.

**Comment by keradus** [comment]
Great to hear @nicolas-grekas 
That is how CS fixes usually works here ;) fabbot.io will catch more changes in PRs if needed.

**Pull request merged**